docs/security/access-restriction-with-fail2ban.md

```diff
@@ -0,0 +1,325 @@
+
+# Prerequisites
+
+Ensure your system is up to date before starting. 
+
+For CentOS, run:
+
+	[root@server ~]# yum update
+
+For Debian, run:
+
+	[root@server ~]# apt-get update; apt-get upgrade
+
+## CentOS 
+
+Fail2ban is provided by the EPEL repository, so this needs to be installed first:
+
+	[root@server ~]# yum -y install epel-release
+
+Update the yum repository and install fail2ban:
+
+	[root@server ~]# yum update; yum -y install fail2ban
+
+## Debian
+
+fail2ban is already included in the standard distribution, so you just need to run:
+
+	[root@server ~]# apt-get install fail2ban
+
+The configuration set for a specific application is called a "jail" and contains the jail settings, the filter and the required action. Fail2ban will permanently monitor the specified log file using the filter and jail settings and trigger the action if the conditions are matched. After a configurable time, another action can be performed (such as unbanning the offending IP).
+
+# Fail2ban configuration
+
+## Configuration files
+
+Configuration files are located under `/etc/fail2ban`. You will find here:
+
+- `jail.conf`: the stock jails configuration, which for most users will be sufficient. Should you wish to create your own jails or adjust the existing jails' configuration, it's recommended to deploy them in a "jail.local" file and leave the original file untouched. Otherwise, your jail.conf customizations might get lost when upgrading fail2ban.
+- `fail2ban.conf`: the general fail2ban configuration file. Same as above, any customizations should be placed in a separate file named fail2ban.local.
+- `filter.d`: the folder containing built-in (predefined) filters, each filter being in a separate file.
+- `action.d`: the folder containing built-in (predefined) actions, each action being in a separate file.
+
+## Starting and stopping fail2ban
+
+The fail2ban service is started/stopped/checked using the standard commands:
+
+	[root@server ~]# /etc/init.d/fail2ban status
+	fail2ban-server is stopped
+	[root@server ~]# /etc/init.d/fail2ban start
+	Starting fail2ban:                                         [  OK  ]
+	[root@server ~]# /etc/init.d/fail2ban status
+	fail2ban-server (pid  763) is running...
+	Status
+	|- Number of jail:      1
+	`- Jail list:           ssh-iptables
+	[root@server ~]# /etc/init.d/fail2ban stop
+	Stopping fail2ban:                                         [  OK  ]
+
+You can see the `ssh-iptables` jail is already active - it monitors the SSH login attempts and uses iptables to ban IPs which fail to authenticate properly.
+
+## Analysing an existing jail 
+
+Let's take a look at an existing jail. Use your favorite file editor, open the `/etc/fail2ban/jail.conf` file and look for the `ssh-iptables` jail configuration. You should find a code block similar to this:
+
+	[ssh-iptables]
```

Comment:
FYI they seem to know split this into two sections based on IPv4 and v6. This is now `[ssh-iptables-ipset4]`.